### PR TITLE
Store creation MVP: site picker entry point behind a feature flag

### PR DESCRIPTION
**Pull request by jaclync**
Part of #7879

### Description

Previously in pe5sF9-Jv-p2, we decided to release the first iteration of store creation in the specified implementation. This PR is the very first part of the MVP implementation, after many rounds of testing (which is time-consuming having to go through 4 profiler questions and a 50% failure rate on my end). Please watch out for the issues in pe5sF9-JD-p2#comment-1124, and let me know if you also see the first two issues during your testing:
- Getting stuck after selecting the WPCOM plan, which happens to me half of the time. In the success case, it navigates to the domain selection page
- The domain selection page is at an unexpected zoom scale, which occurs when I tested in an iOS 16 simulator but not in a physical iOS 15 device

My suspicion is that because the WPCOM plan CTA opens a new window and requires a workaround https://github.com/woocommerce/woocommerce-ios/commit/f323544bf3d9b14f6e59aa7fc9b65158c42bee41, I thought I should look into this issue separately if it's also happening to anyone else in iOS. I can try opening the URL in a new view instead of loading it in the same view.

Please also refer to the subtasks in https://github.com/woocommerce/woocommerce-ios/issues/7879 for any followup tasks, and suggest any improvements that aren't in the list.

#### Delays & retries

As you can see in `StorePickerViewController`, many retries with a delay in between is there because it takes an indefinite amount of time for the newly created site to be available in the WPCOM `/me/sites` API response also as a **Jetpack site**. The new site first appears with `jetpack` and `jetpack_connection` as `false`, then they gradually become `true` 😅 Please feel free to suggest a better way to handle this.

#### Unit testing

I decided to leave unit testing for now, because we're implementing the second iteration in pe5sF9-Ky-p2 right after MVP and honestly I find this iteration quite hacky and potentially unreliable.

#### Feature flag on

- Launch the app
- Log out if needed
- Log in to the app --> when reaching the store picker, there should be a third CTA to create a store
- Tap "Create a new store" --> it should launch a view to create a store, starting with profiler questions and then selecting the WPCOM eCommerce plan. At the end of the flow, it might stay on the success page for a while before the modal is dismissed automatically and then the app should be logged in to the new store
- Go to the Menu tab, and tap "Switch store" --> there should be a second CTA to create a store
- Repeat the step to tap "Create a new store" --> at the end, the app should be switched to the new store

#### Feature flag off

- Disable the feature flag in `DefaultFeatureFlagService`
- Launch the app
- Log out if needed
- Log in to the app --> when reaching the store picker, there should **not** be a third CTA to create a store
- Continue with a store
- Go to the Menu tab, and tap "Switch store" --> there should **not** be a second CTA to create a store

### Screenshots

after logging in | after switching stores
-- | --
IMG_3501 | IMG_3503

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/Store Creation/StoreCreationWebViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            // A successful URL looks like `https://wordpress.com/checkout/thank-you/{{site_url}}/.*`.
+            // There is usually more than one URL requests like this, with different parameters.
+            let regex = try NSRegularExpression(pattern: "\(Constants.completionURLPrefix)" + #"(?<siteURL>[^/]+)"#, options: [])
+            let urlRange = NSRange(location: 0, length: url.count)
+            let matches = regex.matches(in: url, options: [], range: urlRange)
+            guard let match = matches.first else {
+                throw StoreCreationError.invalidCompletionPath
+            }
+
+            // Extracts the site URL substring matching the named capture group `siteURL` in the regex.
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/Store Creation/StoreCreationWebViewModel.swift]: Nit: Would it make sense to move the site URL extraction code to a private method?

File: WooCommerce/Classes/Extensions/Publisher+Concurrency.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+extension Publisher {
+    /// Transforms the publisher with an async throwable operator.
+    ///
+    /// Original implementation:
+    /// https://www.swiftbysundell.com/articles/calling-async-functions-within-a-combine-pipeline
+    ///
+    /// - Parameter transform: a function that transforms the upstream publisher asynchronously with the option to throw an error.
+    /// - Returns: a new publisher that is transformed by the given operator asynchronously.
+    func asyncMap<T>(_ transform: @escaping (Output) async throws -> T) ->
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Extensions/Publisher+Concurrency.swift:11]: TIL 🙇

**Review by selanthiraiyan — APPROVED** [review]
**Testing**

- I was able to successfully create a store. ✅ 

- >Getting stuck after selecting the WPCOM plan, which happens to me half of the time. In the success case, it navigates to the domain selection page

  I faced this problem as well. 

- Thanks for covering the subtasks [here](https://github.com/woocommerce/woocommerce-ios/issues/7879). 👍 

**Code**

I left a non-blocking nit. ✅ 

LGTM :shipit:

File: WooCommerce/Classes/Authentication/Store Creation/StoreCreationWebViewModel.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/Authentication/Store Creation/StoreCreationWebViewModel.swift]: Updated in https://github.com/woocommerce/woocommerce-ios/pull/7886/commits/9139d171fa69b9dadb9a1626b2700f8f9084cac9

**Pull request merged**